features/events/traps/blueprint-trapd-listener.xml

```diff
@@ -11,13 +11,39 @@
         http://aries.apache.org/schemas/blueprint-ext/blueprint-ext-1.1.xsd
 ">
 
-    <cm:property-placeholder id="trapdProperties" persistent-id="org.opennms.netmgt.trapd" update-strategy="reload">
+    <cm:property-placeholder id="trapHandlerDefaultProperties" persistent-id="org.opennms.netmgt.trapd" update-strategy="none">
         <cm:default-properties>
             <cm:property name="trapd.listen.interface" value="127.0.0.1" />
             <cm:property name="trapd.listen.port" value="162" />
         </cm:default-properties>
     </cm:property-placeholder>
 
+    <!-- Camel Event for unmarshalling processor -->
+    <bean id="unmarshaller" class="org.opennms.core.camel.JaxbUtilsUnmarshalProcessor">
+        <argument value="org.opennms.netmgt.config.trapd.TrapdConfiguration" />
+    </bean>
+
+    <camelContext id="trapConnectionHandlerDefaultContext" xmlns="http://camel.apache.org/schema/blueprint">
+        <propertyPlaceholder id="minionproperties" location="blueprint:minionProperties" />
+        <route id="restClientToTrapdConfig">
+            <from uri="timer://restTrapConfigurationTimer?fixedRate=true&amp;period=60000"/>
+            <to uri="{{http-url}}/rest/config/trapd?authMethod=Basic&amp;authUsername={{username}}&amp;authPassword={{password}}" />
+            <bean ref="unmarshaller" />
+            <to uri="bean:trapdConfig?method=onUpdate" />
+            <bean ref="trapReceiver" method="setTrapdConfig" />
```

Comment:
The idea would be to add calls to this interface and class to retrieve the data. This will mean that we won't have to store HTTP credentials in clear text on the system and we'll have a central service for REST access.

https://github.com/OpenNMS/opennms/blob/develop/features/minion/core/impl/src/main/java/org/opennms/minion/core/impl/ScvEnabledRestClientImpl.java